base/string_utils.hpp

```diff
@@ -318,30 +318,30 @@ void ForEachMatched(string const & s, regex const & regex, TFn && fn)
 // form is chosen to fit both std::string and strings::UniString.
 // This function does not normalize either of the strings.
 template <typename TIter>
-uint32_t EditDistance(TIter const & b1, TIter const & e1, TIter const & b2, TIter const & e2)
+size_t EditDistance(TIter const & b1, TIter const & e1, TIter const & b2, TIter const & e2)
 {
   size_t n = distance(b1, e1);
   size_t m = distance(b2, e2);
-  vector<vector<uint32_t>> best(n + 1, vector<uint32_t>(m + 1, n + m));
-  best[0][0] = 0;
-  for (size_t i = 1; i <= n; ++i)
-    best[i][0] = i;
-  for (size_t j = 1; j <= m; ++j)
-    best[0][j] = j;
+  vector<size_t> prev(m + 1);
+  vector<size_t> cur(m + 1);
+  for (size_t j = 0; j <= m; j++)
```

Comment:
Again, no. The meaning of j is the size of string's prefix. It's better to explicitly declare that we're interested in prefixes of sizes from 0 to m inclusively.